pkg/api/v20170701/types.go

```diff
@@ -139,9 +139,18 @@ func (m *MasterProfile) UnmarshalJSON(b []byte) error {
 		m.Count = 1
 	}
 
-	// TODO:
-	// Need to set default values for
-	// OSDiskSizeGB, FirstConsecutiveStaticIP, StorageProfile
+	if m.FirstConsecutiveStaticIP == "" {
+		// if FirstConsecutiveStaticIP is missing, set to default 10.240.255.5
+		m.FirstConsecutiveStaticIP = "10.240.255.5"
+	}
+
+	if m.StorageProfile == "" {
+		// if StorageProfile is missing, set to default StorageAccount
+		m.StorageProfile = StorageAccount
+	}
+
+	// OSDiskSizeGB
+	// The current behavior is, when OSDiskSizeGB is 0, we just don't pass it in the arm template
```

Comment:
OSDIskSizeGB is an override value. vm sizes have default OS disk sizes. We should leave it at 0 if it isn't set and the user should get the default for the vm size.